edi_oca/models/edi_backend.py

```diff
@@ -398,6 +398,19 @@ def _cron_check_output_exchange_sync(self, **kw):
         for backend in self:
             backend._check_output_exchange_sync(**kw)
 
+    def exchange_generate_send(self, recordset, skip_generate=False, skip_send=False):
+        for rec in recordset:
+            if skip_generate:
+                job1 = rec
+            else:
+                job1 = rec.delayable().action_exchange_generate()
+            if hasattr(job1, "on_done"):
```

Comment:
Why do you need the on_done check?

Reply:
I want to check if my record is not a job then it will skip delay

Reply:
maybe add a comment?

Reply:
I will :)

Reply:
I refactored here [`66797c2` (#1035)](https://github.com/OCA/edi/pull/1035/commits/66797c2a1c0607da7a60fe05173237fd80ec02f5) in a way that does not require any comment or check on attributes